torch_xla/csrc/elementwise.cpp

```diff
@@ -128,16 +128,17 @@ xla::XlaOp BuildHardSwishBackward(xla::XlaOp grad_output, xla::XlaOp input) {
   return xla::Select(xla::Ge(input, three), grad_output, stepone);
 }
 
-xla::XlaOp BuildSoftshrink(xla::XlaOp input, const at::Scalar& lambda) {
-  xla::XlaBuilder* builder = input.builder();
-  const xla::Shape& shape = XlaHelpers::ShapeOfXlaOp(input);
-  xla::XlaOp zero = xla::Zero(input.builder(), shape.element_type());
-  xla::XlaOp xla_lambd =
-      XlaHelpers::ScalarBroadcast(lambda.to<double>(), shape, builder);
-  xla::XlaOp le_lambda_branch =
-      xla::Select(xla::Lt(input, Neg(xla_lambd)), input + xla_lambd, zero);
-  return xla::Select(xla::Le(input, xla_lambd), le_lambda_branch,
-                     input - xla_lambd);
+xla::XlaOp BuildSoftshrink(xla::XlaOp input, xla::XlaOp lambda) {
+  const xla::Shape& input_shape = XlaHelpers::ShapeOfXlaOp(input);
```

Comment:
Curious how I can convert a xla::XlaOp object to a scalar? I asked because the lambda for Softshrink needs to be nonnegative (https://pytorch.org/docs/stable/generated/torch.nn.Softshrink.html). How can I do the check here? Also, if I can manage to check lambda here, the check will be done before we actually execute the graph, right?